Rendering/src/Renderer.cpp

```diff
@@ -48,15 +90,47 @@ void Renderer::Update()
 
      // Kinda okay methodology
      //TODO: Replace with Scene based or Material based Draw
-    testMesh.Draw();
-
+    
+
+    // MESHES ============================
+    auto meshComponentUUIDs = EntityManager::getInstance().findEntitiesByComponent(ComponentType::Mesh);
+    
+    for (auto& uuid : meshComponentUUIDs) 
+    {
+        auto entity = EntityManager::getInstance().getEntity(uuid);
+        MeshComponent* meshComponent = dynamic_cast<MeshComponent*>(entity->getComponent(ComponentType::Mesh));
+
+        if (meshComponent != nullptr)
+        {
+            glm::vec3 localScale = entity->transform->getLocalScale();
+
+            glm::mat4 modelMatrix = glm::translate(glm::mat4(1.0f), entity->transform->getWorldPosition()) *
+                                    glm::mat4_cast(entity->transform->getLocalOrientation()) *
+                                    glm::scale(glm::mat4(1.0f), localScale);
+
+            mEngineUniformBuffer.updateMatrices(
```

Comment:
Redundant update. 
We can set the camera UBO before we enter the loop. 
After we can change the model via SetSubData() like on line 62.